lasagne/layers/noise.py

```diff
@@ -15,12 +17,58 @@
 
 
 class DropoutLayer(Layer):
-    def __init__(self, incoming, p=0.5, rescale=True, **kwargs):
-        super(DropoutLayer, self).__init__(incoming, **kwargs)
+    """Dropout layer [1]_,[2]_
+
+    Sets values to zero with probability p. See notes for disabling dropout
+    during testing.
+
+    Parameters
+    ----------
+    input_layer : `Layer` instance
+        The layer from which this layer will obtain its input
+    p : float or scalar tensor
+        The probability of setting a value to zero
```

Comment:
`input_layer` should still be `incoming`, until we decide to not allow creating layers from shape tuples any longer. You can copy the description from the base class. `rescale` is missing in the parameter list.